pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -1225,40 +1220,20 @@ func addMetadataByPrecedence(metadata map[string]string, addedMetadata map[strin
 	}
 }
 
-func (c *Reconciler) updateLabelsAndAnnotations(ctx context.Context, pr *v1beta1.PipelineRun) (*v1beta1.PipelineRun, error) {
-	ctx, span := c.tracerProvider.Tracer(TracerName).Start(ctx, "updateLabelsAndAnnotations")
-	defer span.End()
-	newPr, err := c.pipelineRunLister.PipelineRuns(pr.Namespace).Get(pr.Name)
-	if err != nil {
-		return nil, fmt.Errorf("error getting PipelineRun %s when updating labels/annotations: %w", pr.Name, err)
-	}
-	if !reflect.DeepEqual(pr.ObjectMeta.Labels, newPr.ObjectMeta.Labels) || !reflect.DeepEqual(pr.ObjectMeta.Annotations, newPr.ObjectMeta.Annotations) {
-		// Note that this uses Update vs. Patch because the former is significantly easier to test.
-		// If we want to switch this to Patch, then we will need to teach the utilities in test/controller.go
-		// to deal with Patch (setting resourceVersion, and optimistic concurrency checks).
-		newPr = newPr.DeepCopy()
-		// Properly merge labels and annotations, as the labels *might* have changed during the reconciliation
-		newPr.Labels = kmap.Union(newPr.Labels, pr.Labels)
-		newPr.Annotations = kmap.Union(newPr.Annotations, pr.Annotations)
-		return c.PipelineClientSet.TektonV1beta1().PipelineRuns(pr.Namespace).Update(ctx, newPr, metav1.UpdateOptions{})
-	}
-	return newPr, nil
-}
-
-func storePipelineSpecAndMergeMeta(ctx context.Context, pr *v1beta1.PipelineRun, ps *v1beta1.PipelineSpec, meta *resolutionutil.ResolvedObjectMeta) error {
+func storePipelineSpec(ctx context.Context, pr *v1beta1.PipelineRun, ps *v1beta1.PipelineSpec, meta *resolutionutil.ResolvedObjectMeta) error {
 	// Only store the PipelineSpec once, if it has never been set before.
 	if pr.Status.PipelineSpec == nil {
 		pr.Status.PipelineSpec = ps
 		if meta == nil {
 			return nil
 		}
 
-		// Propagate labels from Pipeline to PipelineRun. PipelineRun labels take precedences over Pipeline.
-		pr.ObjectMeta.Labels = kmap.Union(meta.Labels, pr.ObjectMeta.Labels)
-		pr.ObjectMeta.Labels[pipeline.PipelineLabelKey] = meta.Name
-
-		// Propagate annotations from Pipeline to PipelineRun. PipelineRun annotations take precedences over Pipeline.
-		pr.ObjectMeta.Annotations = kmap.Union(meta.Annotations, pr.ObjectMeta.Annotations)
+		// // Propagate labels from Pipeline to PipelineRun. PipelineRun labels take precedences over Pipeline.
```

Comment:
These can just be dropped?

Reply:
From there yes, overall no as written in the PR description.

> Even if this PR propose to remove this "feature", we still want the same behavior (aka a `Pod` gets its labels from both the `Task` and the `TaskRun`). Because we are only "caching" the spec of the `Task` in `TaskRun.status` (and same goes for `PipelineRun`), we would have to query the metadata for the `Task` each time. Using informers, it wouldn't cost that much, but we would run the same possible race as we did before and that's the reason we use `TaskRun.status` and `PipelineRun.status` as source of truth for the spec in all reconciler loop for a given object. 
> 
> So the question is the following : **Where should we store the `Task` (and `Pipeline` for `PipelineRun`) labels and annotations in a `TaskRun` object ?** Because we have already release `v1`, we can only do "additive" changes, so, most likely, the only way would be to have a `taskMetadata` field in the `status` in addition to `taskSpec`.
But I am curious if anyone has any other ideas 👼🏼.